tests/test_cases/test_struct/test_struct.py

```diff
@@ -49,6 +52,32 @@ async def test_packed_struct_setting(dut):
 
     assert str(dut.my_struct.value) == "000"
 
+    if SIM_NAME.startswith("verilator"):
+        return
+
+    assert dut.my_struct.val_a.value == 0
+    assert dut.my_struct.val_b.value == 0
+    assert dut.my_struct["value"].value == 0
+
+    # test logic object fields
+    dut.my_struct.set(0)
+    dut.my_struct.setimmediatevalue(0)
+    assert dut.my_struct.is_const is False
+
+    # test individual signals -> struct
+
+    dut.my_struct.val_a.value = 1
+    dut.my_struct["value"].value = 1
+
+    await Timer(1000, "ns")
+
+    assert str(dut.my_struct.value) == "101"
+    assert dut.my_struct.val_a.value == 1
+    assert dut.my_struct.val_b.value == 0
+    assert dut.my_struct["value"].value == 1
+
+    assert len(dut.my_struct) == 3
+
```

Comment:
Let's put these in a separate test and set as `expect_error=AttributeError`.

Reply:
As in a separate test, as well as a separate PR to go in first?

Reply:
Oh I see now, nvm I'll leave as one PR